python/samples/concepts/agents/step3_openai_assistant_agent.py

```diff
@@ -0,0 +1,95 @@
+
+HOST_NAME = "Host"
+HOST_INSTRUCTIONS = "Answer questions about the menu."
+
+# Note: you may toggle this to switch between AzureOpenAI and OpenAI
+use_azure_openai = True
+
+
+# Define a sample plugin for the sample
+class MenuPlugin:
+    """A sample Menu Plugin used for the concept sample."""
+
+    @kernel_function(description="Provides a list of specials from the menu.")
+    def get_specials(self) -> Annotated[str, "Returns the specials from the menu."]:
+        return """
+        Special Soup: Clam Chowder
+        Special Salad: Cobb Salad
+        Special Drink: Chai Tea
+        """
+
+    @kernel_function(description="Provides the price of the requested menu item.")
+    def get_item_price(
+        self, menu_item: Annotated[str, "The name of the menu item."]
+    ) -> Annotated[str, "Returns the price of the menu item."]:
+        return "$9.99"
+
+
+# A helper method to invoke the agent with the user input
+async def invoke_agent(agent: OpenAIAssistantAgent, thread_id: str, input: str) -> None:
+    """Invoke the agent with the user input."""
+    await agent.add_chat_message(thread_id=thread_id, message=ChatMessageContent(role=AuthorRole.USER, content=input))
+
+    print(f"# {AuthorRole.USER}: '{input}'")
+
+    async for content in agent.invoke(thread_id=thread_id):
+        if content.role != AuthorRole.TOOL:
+            print(f"# {content.role}: {content.content}")
+
+
+async def main():
+    # Create the instance of the Kernel
+    kernel = Kernel()
+
+    service_id = "agent"
+
+    # Add the sample plugin to the kernel
+    kernel.add_plugin(plugin=MenuPlugin(), plugin_name="menu")
+
+    # Create the agent
+    if use_azure_openai:
+        agent = AzureOpenAIAssistantAgent(
+            kernel=kernel, service_id=service_id, name=HOST_NAME, instructions=HOST_INSTRUCTIONS
+        )
+    else:
+        agent = OpenAIAssistantAgent(
+            kernel=kernel, service_id=service_id, name=HOST_NAME, instructions=HOST_INSTRUCTIONS
+        )
+
+    # Next create the assistant
+    await agent.create_assistant()
+
+    # Note: the agent creation can be done in one step if desired
+    # if use_azure_openai:
+    #     agent = await AzureOpenAIAssistantAgent.create(
+    #         kernel=kernel, service_id=service_id, name=HOST_NAME, instructions=HOST_INSTRUCTIONS
+    #     )
+    # else:
+    #     agent = await OpenAIAssistantAgent.create(
+    #         kernel=kernel, service_id=service_id, name=HOST_NAME, instructions=HOST_INSTRUCTIONS
+    #     )
+
+    thread_id = await agent.create_thread()
+
+    try:
+        # await invoke_agent(agent, thread_id=thread_id, input="Hello")
+        await invoke_agent(agent, thread_id=thread_id, input="What is the special soup?")
+        await invoke_agent(agent, thread_id=thread_id, input="What is the special drink?")
+        await invoke_agent(agent, thread_id=thread_id, input="Thank you")
+    finally:
+        await agent.delete_thread(thread_id)
```

Comment:
same for this, deleting both a thread and assistant?

Reply:
Will potentially provide convenience method in the next iteration.